sys-auth/sssd/sssd-2.3.1.ebuild

```diff
@@ -12,32 +14,35 @@ KEYWORDS="~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~sparc ~x
 
 LICENSE="GPL-3"
 SLOT="0"
-IUSE="acl +autofs +locator +netlink nfsv4 nls +manpages samba selinux +sudo +ssh systemd test valgrind"
+IUSE="acl +autofs doc kcm +locator +netlink nfsv4 nls +manpages pac python samba secrets selinux sudo ssh systemd test valgrind"
 RESTRICT="!test? ( test )"
-REQUIRED_USE="test? ( ssh sudo )"
 
-COMMON_DEP="
+REQUIRED_USE="pac? ( samba )
+	kcm? ( systemd )
+	python? ( ${PYTHON_REQUIRED_USE} )
+	secrets? ( systemd )
+	test? ( ssh sudo )"
+
+DEPEND="
 	>=sys-libs/pam-0-r1[${MULTILIB_USEDEP}]
```

Comment:
I'm a bit surprised that e.g. pam is a multilib dep, but openssl isn't. Can you verify which do/don't need the multilib dep?

Reply:
the only multilib files in this package are the plugins for pam, nss and kerberos, i *think* some binary only stuff segfaulted when i was using nss-pam-ldapd and it wasn't multilib but it has been a few years since that happened.

Reply:
Seems multilib(32bits) now needs samba? Didn't need that before:
```
checking for uintptr_t... yes
checking for ptrdiff_t... yes
checking for nscd... /usr/sbin/nscd
checking for nscd... yes
checking for nsupdate... /usr/bin/nsupdate
checking for executable nsupdate... yes
checking for nsupdate 'realm' support'... yes
checking keyutils.h usability... yes
checking keyutils.h presence... yes
checking for keyutils.h... yes
checking for add_key in -lkeyutils... yes
checking for sigprocmask... yes
checking for sigblock... yes
checking for sigaction... yes
checking for getpgrp... yes
checking for prctl... yes
checking for NDR_NBT... no
configure: error: Please install Samba 4 NDR NBT development libraries.
Samba 4 libraries are necessary for building ad and ipa provider.
If you do not want to build these providers it is possible to build SSSD
without them. In this case, you will need to execute configure script
with argument --without-samba

Reply:
good catch, try this (against tree version):
```
--- sssd-2.3.1.ebuild.orig	2020-08-25 01:21:11.538985183 +0200
+++ sssd-2.3.1.ebuild	2020-08-25 20:03:57.485186910 +0200
@@ -50,7 +50,7 @@
 	nls? ( >=sys-devel/gettext-0.18 )
 	pac? (
 		app-crypt/mit-krb5[${MULTILIB_USEDEP}]
-		net-fs/samba[${MULTILIB_USEDEP}]
+		net-fs/samba
 	)
 	python? ( ${PYTHON_DEPS} )
 	samba? ( >=net-fs/samba-4.10.2[winbind] )
@@ -188,6 +188,7 @@
 			# ldb lib fails... but it does not seem to bother
 			{DHASH,COLLECTION,INI_CONFIG_V{0,1,1_1,1_3}}_{CFLAGS,LIBS}=' '
 			{PCRE,CARES,SYSTEMD_LOGIN,SASL,GLIB2,DBUS,CRYPTO,P11_KIT}_{CFLAGS,LIBS}=' '
+			{NDR_NBT,SMBCLIENT,NDR_KRB5PAC}_{CFLAGS,LIBS}=' '
 
 			# use native include path for dbus (needed for build)
 			DBUS_CFLAGS="${native_dbus_cflags}"
@@ -196,6 +197,7 @@
 			ac_cv_lib_ldap_ldap_search=yes
 			--without-secrets
 			--without-kcm
+			--with-smb-idmap-interface-version=6
 		)
 	fi
 
```

Reply:
If that works for Joakim, please submit a pull request and mention me. (Also include a patch to remove the `pac` USE flag if you like)

Reply:
That builds but I am not happy with that hack. Why not have --without-samba instead of
--with-smb-idmap-interface-version=6
NDR_NBT,SMBCLIENT,NDR_KRB5PAC}_{CFLAGS,LIBS}=' '
hacks ?

Also, I have always used --with-crypto="" too in my ebuild, nothing in 32bits seems
to need openssl, does it?

Not sure about sssd_krb5_locator_plugin.so either, can it be used for 32 bits?

USE=manpages should probably be USE=man instead

Reply:
hmm, I note that with your fix, 32 bits pac builds so perhaps use that anyway.
I do think openssl is unwated, adding --with-crypto="" seems to work, at least for me.

Reply:
Finally, a -9999 version would be appreciated as sssd is slow to release.
I use this patch to avoid the famous samba lib bug: 
```
--- ./src/external/samba.m4.org	2020-06-25 11:04:18.221147249 +0200
+++ ./src/external/samba.m4	2020-06-25 11:05:04.654862849 +0200
@@ -60,7 +60,7 @@
         SAVE_CFLAGS=$CFLAGS
         SAVE_LIBS=$LIBS
         CFLAGS="$CFLAGS $SMBCLIENT_CFLAGS $NDR_NBT_CFLAGS $NDR_KRB5PAC_CFLAGS"
-        LIBS="$LIBS -L${sambalibdir} -lidmap-samba4 -Wl,-rpath ${sambalibdir}"
+        LIBS="$LIBS -L${sambalibdir} -Wl,--no-as-needed -lidmap-samba4 -lreplace-samba4 -lsmbd-shim-samba4 -Wl,-rpath ${sambalibdir}"
         AC_RUN_IFELSE(
             [AC_LANG_SOURCE([
 #include <stdlib.h>

Reply:
> That builds but I am not happy with that hack. Why not have --without-samba instead of
> --with-smb-idmap-interface-version=6
> NDR_NBT,SMBCLIENT,NDR_KRB5PAC}_{CFLAGS,LIBS}=' '
> hacks ?
because the pac responder doesn't build with --without-samba, if upstream provided a way to build the krb and pam plugins separately this mess would not be required but i don't think there is much of a use-case for them.
don't really want to patch configure either.


> Also, I have always used --with-crypto="" too in my ebuild, nothing in 32bits seems
> to need openssl, does it?

there is no point HAVE_LIBCRYPTO is still 1 with that.

> Not sure about sssd_krb5_locator_plugin.so either, can it be used for 32 bits?
probably. i guess my earlier comment still applies but i don't use any of the kerberos plugins myself

> USE=manpages should probably be USE=man instead
yeah, probably (why is man even a local useflag still at this point?)

> I use this patch to avoid the famous samba lib bug
its not famous enough that i know about it. what exactly does that fix?

Reply:
> > That builds but I am not happy with that hack. Why not have --without-samba instead of
> > --with-smb-idmap-interface-version=6
> > NDR_NBT,SMBCLIENT,NDR_KRB5PAC}_{CFLAGS,LIBS}=' '
> > hacks ?
> 
> because the pac responder doesn't build with --without-samba, if upstream provided a way to build the krb and pam plugins separately this mess would not be required but i don't think there is much of a use-case for them.
> don't really want to patch configure either.

Yes, I noticed that later.

> 
> > Also, I have always used --with-crypto="" too in my ebuild, nothing in 32bits seems
> > to need openssl, does it?
> 
> there is no point HAVE_LIBCRYPTO is still 1 with that.

hmm, that is a bit odd. 

> 
> > Not sure about sssd_krb5_locator_plugin.so either, can it be used for 32 bits?
> 
> probably. i guess my earlier comment still applies but i don't use any of the kerberos plugins myself
> 
> > USE=manpages should probably be USE=man instead
> 
> yeah, probably (why is man even a local useflag still at this point?)
> 
> > I use this patch to avoid the famous samba lib bug
> 
> its not famous enough that i know about it. what exactly does that fix?

https://bugs.gentoo.org/692800

Reply:
> https://bugs.gentoo.org/692800

ugh. i'd... rather use ```--with-smb-idmap-interface-version=6``` in native as well as we already depend on samba newer then 4.7 anyway (it by itself doesn't generate a dep on samba, at least on non-native but i'll re-test that of course.

Reply:
> > https://bugs.gentoo.org/692800
> 
> ugh. i'd... rather use `--with-smb-idmap-interface-version=6` in native as well as we already depend on samba newer then 4.7 anyway (it by itself doesn't generate a dep on samba, at least on non-native but i'll re-test that of course.

For 32 bits, yes the --with-smb-idmap-interface-version=6 is best.
But the native part will fail everynow and then. Not sure what the I/F is in samba 4.12 ? if still 6 one could use with-smb-idmap-interface-version=6 overall.